be/src/storage/rowset_column_update_state.cpp

```diff
@@ -96,12 +96,17 @@ Status RowsetColumnUpdateState::_load_upserts(Rowset* rowset, uint32_t idx) {
         pk_columns.push_back((uint32_t)i);
     }
     std::vector<uint32_t> update_columns;
+    std::vector<uint32_t> update_columns_with_keys;
     const auto& txn_meta = rowset->rowset_meta()->get_meta_pb().txn_meta();
     for (uint32_t cid : txn_meta.partial_update_column_ids()) {
-        update_columns.push_back(cid);
+        update_columns_with_keys.push_back(cid);
```

Comment:
If remove the keys from the delta group, how to corresponds the delta group with main segment?

Reply:
The rows in delta column is already sorted, and correspond with rows in main segment one by one.